### PR TITLE
Aws s3 integration 2

### PR DESCRIPTION
**Pull request by Vilina**
Change image upload to use AWS S3 for staging and production

### CODE REVIEW

File: src/middleware/image-update-delete/imageUpdateDelete.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
+import {RequestHandler} from "express";
+import imageUpload from "../image-upload/multer";
+import config from '../../config/index';
+import s3 from '../../libs/s3Client';
+import {DeleteObjectCommand} from "@aws-sdk/client-s3";
+import ProductDao from '../../modules/products/dao/ProductDao';
+import ProductModel from "../../modules/products/model/ProductModel";
+
+const deleteImages = async (images: string[]) => {
```
Comment (RubenBarseghyan) [inline src/middleware/image-update-delete/imageUpdateDelete.ts]: Suggestion: Wrap try catch block

File: src/middleware/image-update-delete/imageUpdateDelete.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        }));
+    })
+)};
+
+const imageUpdateDelete: RequestHandler = async (req, res, next) => {
+    const productDao = new ProductDao(ProductModel);
+    const oldProductData = await productDao.findProductById(req.params.id);
+
+    if(req.method === 'PUT') {
+        console.log("Updating images");
```
Comment (RubenBarseghyan) [inline src/middleware/image-update-delete/imageUpdateDelete.ts:23]: question: Do we need logs?

File: src/modules/products/controller/productController.ts [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+export const getProductById = async (req: Request, res: Response): Promise<void> => {
+    try {
+        const productDao = new ProductDao(ProductModel)
+        const product = await productDao.findProductById(req.params.id);
+        if (product) {
+            product.imageUrls = [];
+            product.images.map((image: any) => {
+                product.imageUrls?.push(`https://${config.aws.aws_bucket_name}.s3.${config.aws.aws_region}.amazonaws.com/${image}`);
+            });
+            console.log(product, 'product')
```
Comment (RubenBarseghyan) [inline src/modules/products/controller/productController.ts]: question: Do we need console?

File: src/modules/products/controller/productController.ts [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+ * @param req - The request object.
+ * @param res - The response object.
+ */
+export const getProductById = async (req: Request, res: Response): Promise<void> => {
+    try {
+        const productDao = new ProductDao(ProductModel)
+        const product = await productDao.findProductById(req.params.id);
+        if (product) {
+            product.imageUrls = [];
+            product.images.map((image: any) => {
```
Comment (RubenBarseghyan) [inline src/modules/products/controller/productController.ts]: question: type of image is String? maybe could escape from any

File: src/middleware/image-update-delete/imageUpdateDelete.ts [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    return  await Promise.all(images.map(async (key: string) => {
+        return await s3.send(new DeleteObjectCommand({
+            Bucket: config.aws.aws_bucket_name,
+            Key: key
+        }));
+    })
+)};
+
+const imageUpdateDelete: RequestHandler = async (req, res, next) => {
+    const productDao = new ProductDao(ProductModel);
```
Comment (RubenBarseghyan) [inline src/middleware/image-update-delete/imageUpdateDelete.ts:19]: Suggestion: Wrap try catch block

**Review by RubenBarseghyan — CHANGES_REQUESTED** [review]
We also have conflict file: db_connect.ts

File: src/modules/products/controller/productController.ts (hunk 4 above)
Comment (Vilina) [reply src/modules/products/controller/productController.ts]: ec68572fde36832e71effb99ca4101be308ff384

File: src/modules/products/controller/productController.ts (hunk 3 above)
Comment (Vilina) [reply src/modules/products/controller/productController.ts]: 65dea0e778bc61db288a1a0b0708b1035774cd7d

**Comment by Vilina** [comment]
> We also have conflict file: db_connect.ts

moved that to config to consolidate all env variables in one place, and refactored the code after

File: src/middleware/image-update-delete/imageUpdateDelete.ts (hunk 1 above)
Comment (Vilina) [reply src/middleware/image-update-delete/imageUpdateDelete.ts]: 65dea0e778bc61db288a1a0b0708b1035774cd7d

**Review by RubenBarseghyan — DISMISSED** [review]

**Review by RubenBarseghyan — APPROVED** [review]

**Pull request merged**